sdb/commands/ustacks.py

```diff
@@ -0,0 +1,280 @@
+                              ret_from_fork+0x35
+            ...
+
+        Print stacks containing the l2arc_feed_thread function
+
+            sdb> stacks -c l2arc_feed_thread
+            TASK_STRUCT        STATE             COUNT
+            ==========================================
+            0xffff9521b3f43b80 INTERRUPTIBLE         1
+                              __schedule+0x24e
+                              schedule+0x2c
+                              schedule_timeout+0x15d
+                              __cv_timedwait_common+0xdf
+                              __cv_timedwait_sig+0x16
+                              l2arc_feed_thread+0x66
+                              thread_generic_wrapper+0x74
+                              kthread+0x121
+                              ret_from_fork+0x35
+
+
+        Count the number of stacks in the zfs module
+
+           sdb> stacks -m zfs | count
+           (unsigned long long)12
+
+    """
+
+    names = ["stacks", "stack"]
+    input_type = "int"
+    output_type = "int"
+    load_on = [sdb.Userland()]
+
+    def __init__(self,
+                 args: Optional[List[str]] = None,
+                 name: str = "_") -> None:
+        super().__init__(args, name)
+        self.func_start, self.func_end = 0, 0
+
+    @classmethod
+    def _init_parser(cls, name: str) -> argparse.ArgumentParser:
+        parser = super()._init_parser(name)
+        parser.add_argument(
+            "-a",
+            "--all",
+            action="store_true",
+            help="list all threads for each unique stack trace" +
+            " instead of printing a single representative thread")
+        parser.add_argument(
+            "-c",
+            "--function",
+            help="only print threads whose stacks contains FUNCTION")
+        return parser
+
+    @staticmethod
+    def get_frame_pcs(thread: Thread) -> List[int]:
+        frame_pcs = []
+        try:
+            for frame in thread.stack_trace():
+                frame_pcs.append(frame.pc)
+        except ValueError:
+            #
+            # Unwinding the stack of a running/runnable task will
+            # result in an exception. Since we expect some tasks to
+            # be running, we silently ignore this case, and move on.
+            #
+            # Unfortunately, the exception thrown in this case is a
+            # generic "ValueError" exception, so we may wind up
+            # masking other "ValueError" exceptions that are not due
+            # to unwinding the stack of a running task.
+            #
+            # We can't check the state of the task here, and verify
+            # it's in the "R" state, since that state can change in
+            # between the point where the "ValueError" exception was
+            # originally raised, and here where we'd verify the
+            # state of the task; i.e. it could have concurrently
+            # transitioned from running to some other state.
```

Comment:
That could be something worth mentioning the `man/help` message above; i.e. "the running thread may not come up on live processes"

BTW is that the case for core dumps too?